### PR TITLE
Add hget, hset, and hgetAll to misk-redis

### PR DESCRIPTION
**Pull request by ddiekroeger**
hget, hgetall, and hset are useful redis operators that let you add a `field` in addition to a `key`. We would like to use this in our service, so I am proposing that we add them here to misk redis.

### CODE REVIEW

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt [hunk 1]
```diff
@@ -47,6 +47,24 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
     }
   }
 
+  // Get a map of field -> value pairs for the given key.
```
Comment (adrw) [inline misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:50]: Could these be proper kotlindoc comments?

```
/** Get a map ... */
```

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt [hunk 2]
```diff
@@ -76,6 +94,13 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
     }
   }
 
+  // Set a ByteArray value for the given key and field.
```
Comment (adrw) [inline misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:97]: kotlin doc style?

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt [hunk 3]
```diff
@@ -47,6 +47,24 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
...
+    jedisPool.resource.use { jedis ->
+      return jedis.hgetAll(key.toByteArray(charset))?.mapKeys {
+        it.key.toString(charset)
+      }?.mapValues {
+        it.value.toByteString()
+      }
+    }
+  }
+
+  // Get a ByteString value for the given key and field
```
Comment (adrw) [inline misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:61]: kotlin doc style?

**Review by adrw — APPROVED** [review]

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt (hunk 1 above)
Comment (ddiekroeger) [reply misk-redis/src/main/kotlin/misk/redis/RealRedis.kt]: good call, updated :)

**Pull request merged**